### ISSUE TITLE
Unable to use player_stats() method with some replays

### ISSUE THREAD

**Issue by TomQWERTY** [pnxenopoulos/awpy#259]
I am trying to parse all my replays and put some data from them to the database. For that purpose I'm using the player_stats() method but for some replays it just doesn't work (an archive containing one of such replays is attached). The program throws an exception:

File "f:\csgo_statistics\test2.py", line 21, in <module>
    all_players = player_stats(data["gameRounds"])
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user1\AppData\Local\Programs\Python\Python311\Lib\site-packages\awpy\analytics\stats.py", line 189, in player_stats
    kast[victim_key]["s"] = False
    ~~~~^^^^^^^^^^^^
KeyError: '76561198128988489'

I can't understand what is the difference between replays that is parsed fine and these defected ones. All of them was played in CS:GO matchmaking and then downloaded using game's menu. When I try to watch such replay using a game client everything is fine. Also when I parse the replay but don't use player_stats() everything is OK too (I can access keys like mapName etc).

I am using Windows 10, Python 3.11.1 if it can be useful.
Not sure it is a place for such questions but I don't know where else to ask it.

[match730_003598853741287047211_1478128697_186.zip](https://github.com/pnxenopoulos/awpy/files/12016247/match730_003598853741287047211_1478128697_186.zip)

**Comment by pnxenopoulos** [comment]
Thanks for raising this issue, @TomQWERTY. Can you try using the most recent awpy (from github, not from `pip install awpy`? You can clone the repo and run `pip install .`

Also, is there anything weird about the demo? E.g., did a player disconnect or something?

**Comment by TomQWERTY** [comment]
> Thanks for raising this issue, @TomQWERTY. Can you try using the most recent awpy (from github, not from `pip install awpy`? You can clone the repo and run `pip install .`
> 
> Also, is there anything weird about the demo? E.g., did a player disconnect or something?

I have tried to use awpy from github but I haven't succeed. I installed it as you said and it was visible through "pip show awpy" but I couldn't use it in a program, I got an error "import could not be resolved".

As for the demo, it really had player disconnection but such disconnections also happened in demos that does not cause this error. Also there were two players who were voted off. I thought that was a reason so I watched more demos that cause this error to find out if they also has players voted off. And many of them really has but not all of them so it doesn't look like a reason of an error.

**Comment by TomQWERTY** [comment]
looks like I managed to fix it by myself.

Firstly, the reason of the problem. player_stats() method breaks if the demo it is dealing with has a player disconnection which happened BEFORE the freezetime ended. When such disconnection happens, player is not stored in gameRound["ctSide"] nor in gameRound["tSide"] (because according to the documentation it stores players who was on certain side at the end of the freeztime). But kast variable row is created only for those players who belong to T or CT. So player who disconnected does not have it. But his death (disconnection is a kill where a player who disconnects is both killer and victim) belongs to this round no matter it happened during freezetime. So a program tries to modify his row in kast variable to state that he didn't survive the round (kast[victim_key]["s"] = False, like in my exception) but as was said above his row does not exist. So we get an exception.

Secondly, the fix. Actually it is very simple. I fixed it just by adding a condition: modify kast only if the victim has one. According to a logic of player_stats() method a player shouldn't have a row in kast variable if he wasn't on some side at the end of the freezetime. So if he doesn't have one, we don't need to modify it. That's why I think that this fix can be used. If I wrong somewhere, please tell me.

This is how it looks now:
if victim_key in kast:
     kast[victim_key]["s"] = False

**Comment by JanEricNitschke** [comment]
Nice find. Although i just tested your demo with the current main branch of awpy and it seems we have already fixed the issue there.

But until we push a new release from that this seems like a good workaround.

**Comment by JanEricNitschke** [comment]
Should be fixed now with the release of awpy 1.3.1
